src/app/auth/authentication.service.ts

```diff
@@ -145,6 +151,108 @@ export class AuthenticationService {
     localStorage.setItem('refresh_token', token.refresh_token);
     return token;
   }
+  
+  connectGitHub(redirectUrl: string) {
+    //let headers = new Headers({ 'Content-Type': 'application/json' });
+    //let options: RequestOptions = new RequestOptions({ headers: headers });
+    let tokenUrl = this.apiUrl + 'token/link';
+    let form = document.createElement("form");
+
+    // Create the for parameter
+    let inputFor = document.createElement("input");
+    inputFor.type = "hidden";
+    inputFor.name = "for";
+    inputFor.value = "https://github.com";
+    form.appendChild(inputFor);
+
+    // Create the token parameter
+    let inputToken = document.createElement("input");
+    inputToken.type = "hidden";
+    inputToken.name = "token";
+    inputToken.value = this.getToken();
+    form.appendChild(inputToken);
+
+    // Create the redirect parameter
+    let inputRedirect = document.createElement("input");
+    inputRedirect.type = "hidden";
+    inputRedirect.name = "redirect";
+    inputRedirect.value = redirectUrl;
+    form.appendChild(inputRedirect);
+
+    form.action = tokenUrl;
+    form.method = "POST";
+    document.body.appendChild(form);
+    form.submit();
+  }
+  
+  disconnectGitHub() {
+    let tokenUrl = this.apiUrl + 'token?for=https://github.com';
+    const xhr = new XMLHttpRequest();
+    xhr.open("delete", tokenUrl);
```

Comment:
The angular http library should be used for requests and not XMLHttpRequest (see this.http.get(tokenUrl, options)  below)